core/kernel.go

```diff
@@ -73,300 +65,305 @@ const (
 // Kernel is the root structure of Burrow
 type Kernel struct {
 	// Expose these public-facing interfaces to allow programmatic extension of the Kernel by other projects
-	Emitter    event.Emitter
-	Service    *rpc.Service
-	Launchers  []process.Launcher
-	State      *state.State
-	Blockchain *bcm.Blockchain
-	Node       *tendermint.Node
-	Transactor *execution.Transactor
-	// Time-based UUID randomly generated each time Burrow is started
-	RunID          simpleuuid.UUID
+	Emitter        event.Emitter
+	Service        *rpc.Service
+	Launchers      []process.Launcher
+	State          *state.State
+	Blockchain     *bcm.Blockchain
+	Node           *tendermint.Node
+	Transactor     *execution.Transactor
+	RunID          simpleuuid.UUID // Time-based UUID randomly generated each time Burrow is started
 	Logger         *logging.Logger
+	txCodec        txs.Codec
+	exeOptions     []execution.ExecutionOption
+	exeChecker     execution.BatchExecutor
+	exeCommitter   execution.BatchCommitter
+	keyClient      keys.KeyClient
+	keyStore       *keys.KeyStore
 	nodeInfo       string
 	processes      map[string]process.Process
 	shutdownNotify chan struct{}
 	shutdownOnce   sync.Once
 }
 
-func NewBurrowDB(dbDir string) dbm.DB {
+func newBurrowDB(dbDir string) dbm.DB {
 	return dbm.NewDB(BurrowDBName, dbm.GoLevelDBBackend, dbDir)
 }
 
-func NewKernel(ctx context.Context, keyClient keys.KeyClient, privValidator tmTypes.PrivValidator,
-	genesisDoc *genesis.GenesisDoc, tmConf *tmConfig.Config, rpcConfig *rpc.RPCConfig, keyConfig *keys.KeysConfig,
-	keyStore *keys.KeyStore, exeOptions []execution.ExecutionOption, authorizedPeersProvider abci.PeersFilterProvider, restore string, nodeKey *crypto.PrivateKey, logger *logging.Logger) (*Kernel, error) {
-
-	var err error
-	kern := &Kernel{
+// NewKernel initializes an empty kernel
+func NewKernel() (*Kernel, error) {
+	runID, err := simpleuuid.NewTime(time.Now()) // Create a random ID based on start time
+	if err != nil {
+		return nil, err
+	}
+	return &Kernel{
+		RunID:          runID,
 		processes:      make(map[string]process.Process),
 		shutdownNotify: make(chan struct{}),
-	}
-	// Create a random ID based on start time
-	kern.RunID, err = simpleuuid.NewTime(time.Now())
+		txCodec:        txs.NewAminoCodec(),
+	}, nil
+}
 
-	logger = logger.WithScope("NewKernel()").With(structure.TimeKey, log.DefaultTimestampUTC,
-		structure.RunId, kern.RunID.String())
-	tendermintLogger := logger.With(structure.CallerKey, log.Caller(LoggingCallerDepth+1))
+// SetLogger initializes the kernel with the provided logger
+func (kern *Kernel) SetLogger(logger *logging.Logger) {
+	logger = logger.WithScope("NewKernel()").With(structure.TimeKey,
+		log.DefaultTimestampUTC, structure.RunId, kern.RunID.String())
 	kern.Logger = logger.WithInfo(structure.CallerKey, log.Caller(LoggingCallerDepth))
-	stateDB := NewBurrowDB(tmConf.DBDir())
+	kern.Emitter = event.NewEmitter(kern.Logger)
```

Comment:
This needs to be part of `NewKernel` - otherwise we do not have a working Kernel just because we haven't set up logger. Actually surprised this has not broken tests...

This means we probably need a `SetLogger` method on `Emitter`. So lets' drop `logger` from the constructor.